### PR TITLE
ESQL: Move expression translators

### PR DESCRIPTION
**Pull request by nik9000**
This moves `ExpressionTranslator` implementations that we use from esql-core into esql. It moves the classes into their own file and deletes the translators we don't currently use.

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsqlExpressionTranslators.java [hunk 1]
```diff
@@ -77,455 +36,24 @@ public final class EsqlExpressionTranslators {
...
         new InComparisons(),
         new Ranges(), // Create Range in PushFiltersToSource for qualified pushable filters on the same field.
-        new ExpressionTranslators.BinaryLogic(),
-        new ExpressionTranslators.IsNulls(),
-        new ExpressionTranslators.IsNotNulls(),
-        new ExpressionTranslators.Nots(),
-        new ExpressionTranslators.Likes(),
-        new ExpressionTranslators.StringQueries(),
-        new ExpressionTranslators.Matches(),
-        new ExpressionTranslators.MultiMatches(),
```
[the remark this answers is not in the document]
Comment (bpintea) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsqlExpressionTranslators.java:87]: I suppose they've just been kept [from before](https://github.com/elastic/elasticsearch/blob/c4758956e9358d5f708cb055e27b8d21f2f3b066/x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/ExpressionTranslators.java#L95), but aren't used (yet?).

**Review by bpintea — APPROVED** [review]
Lgtm

**Review by alex-spies — CHANGES_REQUESTED** [review]
I agree with the refactoring, but think that removing the MultiMatches and StringQueries translators may be better done in a separate PR as it may interfere with ongoing work on MATCH.

Also note that as part of https://github.com/elastic/elasticsearch/issues/106679, we wanted to go over all (used) expression translators and double check if they're actually properly tested. This refactoring is mechanical so it's unlikely to cause a bug, but if it does tests are not 100% going to catch it.

**Comment by nik9000** [comment]
I'm happy either way. I can keep the files.

**Review by alex-spies — APPROVED** [review]
Thanks Nik!

**Pull request closed without merging**